<?xml version="1.0"?>
<launch>
  <node pkg="apriltags_ros" type="apriltag_detector_node" name="apriltag_detector" output="screen" ns="fisheye_undistorted">
    <!-- Remap topic required by the node to custom topics -->
    <remap from="image_rect" to="image_raw" />
    <remap from="camera_info" to="camera_info" />

    <!-- Optional: Subscribe to the compressed stream-->
    <param name="image_transport" type="str" value="raw" />

    <!-- Select the tag family: 16h5, 25h7, 25h9, 36h9, or 36h11(default) -->
    <param name="tag_family" type="str" value="36h11" />

    <!-- Enable projected optical measurements for more accurate tag transformations -->
    <!-- This exists for backwards compatability and should be left true for new setups -->
    <param name="projected_optics" type="bool" value="true" />

    <!-- Describe the tags -->
    <rosparam param="tag_descriptions">[
      {id: 0, size: 0.1762},
      {id: 1, size: 0.1762},
      {id: 2, size: 0.1762},
      {id: 3, size: 0.1762},
      {id: 4, size: 0.1762},
      {id: 5, size: 0.1762},
      {id: 6, size: 0.1762},
      {id: 7, size: 0.1762},
      {id: 8, size: 0.1762},
      {id: 9, size: 0.1762},
      {id: 10, size: 0.1762},
      {id: 11, size: 0.1762},
      {id: 12, size: 0.1762},
      {id: 13, size: 0.1762},
      {id: 14, size: 0.1762},
      {id: 15, size: 0.1762},
      {id: 16, size: 0.1762},
      {id: 17, size: 0.1762},
      {id: 18, size: 0.1762},
      {id: 19, size: 0.1762},
      {id: 20, size: 0.1762},
      {id: 21, size: 0.1762},
      {id: 22, size: 0.1762},
      {id: 23, size: 0.1762},
      {id: 38, size: 0.1725},
      {id: 39, size: 0.1725},
      {id: 584, size: 0.165}]
    </rosparam>
  </node>

  <node pkg="mobility_sensing" type="tag_frames_semantic.py" name="tag_frames_semantic" output="screen"></node>
</launch>
